### PR TITLE
Fixed: turning on your microphone will now take you out of Away mode

### PR DESCRIPTION
**Pull request by amantley**
Previously you could turn on your mic in Away mode.  When you left Away mode, it would be muted again.

https://highfidelity.fogbugz.com/f/cases/21511/If-you-unmute-yourself-while-away-exiting-away-mode-will-mute-you
https://highfidelity.atlassian.net/browse/BUGZ-45

The fix here is to make it so that you become active again if you turn on your mic.

Also, if you are muted and in Away mode, you will now be restored to your previous mute state when you exit Away mode.

### Test Plan
1. open interface
2. with your mic unmuted, press 'esc' to enter away mode,
3. your mic should now be muted.  if you unmute you will exit away mode, your mic will remain unmuted
4. repeat the first three steps, but start with you mic muted. your mic will be unmuted when you exit away mode.
5. with you mic unmuted hit 'esc' to go to away mode with mic muted.  then press a key to return to active.  your mic should be unmuted again.
6. repeat 5 but start with your mic muted.  when you return to active, you mic will be muted once  more

### CODE REVIEW

File: scripts/system/away.js [hunk 1]
```diff
@@ -223,8 +223,8 @@ function setAwayProperties() {
 
 function setActiveProperties() {
     isAway = false;
-    if (!wasMuted) {
-        Audio.muted = !Audio.muted;
+    if ((Audio.muted === true) && (wasMuted === false)) {
```
Comment (hyperlogic) [inline scripts/system/away.js]: I'd rather this be `(Audio.muted && !wasMuted)`

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19107/HighFidelity-Beta-PR15482-c0da464.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19107/HighFidelity-Quest-Beta-PR15482-c0da464.apk)

**Comment by hyperlogic** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19134/HighFidelity-Beta-PR15482-c0da464.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19134/HighFidelity-Quest-Beta-PR15482-c0da464.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19134/HighFidelity-Beta-PR15482-c0da464.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19134/HighFidelity-Beta-Interface-PR15482-c0da464.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19134/HighFidelity-Beta-PR15482-c0da464.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19134/HighFidelity-Beta-Interface-PR15482-c0da464.exe

**Pull request merged**